.github/workflows/cli.yml

```diff
@@ -0,0 +1,70 @@
+name: Viam CLI
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      release_type:
+        required: true
+        type: string
+    secrets:
+      GCP_CREDENTIALS:
+        required: true
+
+jobs:
+  viam-cli:
+    strategy:
+      matrix:
+        include:
+        - goos: linux
+          goarch: amd64
+        - goos: linux
+          goarch: arm64
+        - goos: darwin
+          goarch: amd64
+        - goos: darwin
+          goarch: arm64
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-go@v4
+      with:
+        go-version: '1.20.x'
+
+    - name: Check out code
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
+      uses: actions/checkout@v3
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: build
+      env:
+        GOOS: ${{ matrix.goos }}
+        GOARCH: ${{ matrix.goarch }}
+      run: go build -ldflags="-X 'go.viam.com/rdk/config.Version=${{ github.ref_name }}'" ./cli/viam
+
+    - name: Authorize GCP Upload
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+    - name: upload
+      uses: google-github-actions/upload-cloud-storage@v0.10.4
+      with:
+        headers: "cache-control: no-cache"
+        path: 'viam'
+        destination: 'packages.viam.com/apps/viam-cli/${{ inputs.release_type }}-${{ matrix.goos }}-${{ matrix.goarch }}'
```

Comment:
Please name the binaries themselves (not the folder) using the same ordering we use for viam-server itself. We don't have GOOS in those though, so probably best to add the OS before the arch. So a final filename would look like `viam-cli-latest-linux-x86_64`

As I previously suggested combining all versions into one job, it may make sense to have the renames happen inside a loop doing the builds, and then a wildcard upload can happen.

Reply:
FYI I used GOARCH in the filename (amd64 / arm64) instead of x86_64 / aarch64 because this is slightly more idiomatic in the Makefile